FinSearchUnified/Helper/StaticHelper.php

```diff
@@ -363,7 +363,7 @@ public static function setSmartDidYouMean(SimpleXMLElement $xmlResponse)
         $originalQuery = (string)$query->originalQuery;
         $didYouMeanQuery = (string)$query->didYouMeanQuery;
         $queryString = $query->queryString;
-        $queryStringType = (string)$queryString->attributes()->type;
+        $queryStringType = $queryString->attributes() !== null ? (string)$queryString->attributes()->type : null;
```

Comment:
Why was this changed? It has nothing to do with this story.

Reply:
This was causing the tests to fail as I introduced the demoResponse.xml and it does not have attributes in it. So this was just a precautionary error handling.

Reply:
Changing production code, to make tests work is generally speaking a bad idea. There are only two cases where this change would've been necessary.

1. The response is valid and the code was never working at all. => Adapt the code, so this case is handled properly.
2. The response is invalid and the code threw an exception, because he didn't expect a malformed response. => Adapt the response.

Since nobody has ever complained about this, nor anyone ever had this error, I am pretty sure that the response is invalid or malformed.

Reply:
@TheKeymaster I am not aware of how the response is generated. I only have the demo response which was provided by @dblaichinger and in that the `attributes` was not available. In another case, we were also checking for the `type` of attribute, so based on the information I have, I updated the code to handle either of cases in the response; whether the attributes is there, or not, our code would then not fail.

Reply:
I just checked the response. The change seems to be fine.